bindata/network/ovn-kubernetes/common/ipsec-pre414.yaml

```diff
@@ -0,0 +1,298 @@
+{{if .OVNIPsecDaemonsetEnable}}
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: ovn-ipsec-pre414
+  namespace: openshift-ovn-kubernetes
+  annotations:
+    kubernetes.io/description: |
+      This DaemonSet launches the ovn ipsec networking components for all nodes.
+    release.openshift.io/version: "{{.ReleaseVersion}}"
+spec:
+  selector:
+    matchLabels:
+      app: ovn-ipsec
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 10%
+  template:
+    metadata:
+      annotations:
+        target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+      labels:
+        app: ovn-ipsec
+        component: network
+        type: infra
+        openshift.io/component: network
+        kubernetes.io/os: "linux"
+    spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: network.operator.openshift.io/dpu-host
+                operator: DoesNotExist
+      serviceAccountName: ovn-kubernetes-node
+      hostNetwork: true
+      dnsPolicy: Default
+      priorityClassName: "system-node-critical"
+      initContainers:
+      - name: ovn-keys
+        image: "{{.OvnImage}}"
+        command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+          set -exuo pipefail
+
+          # check if need to run on host
+          rpm -qa libreswan
+          if [ "$?" -eq 0 ]; then
+            echo "host have libreswan, ipsec will run on host"
+            exit 0
+          fi
+
+          # Every time we restart this container, we will create a new key pair if
+          # we are close to key expiration or if we do not already have a signed key pair.
+          #
+          # Each node has a key pair which is used by OVS to encrypt/decrypt/authenticate traffic
+          # between each node. The CA cert is used as the root of trust for all certs so we need
+          # the CA to sign our certificate signing requests with the CA private key. In this way,
+          # we can validate that any signed certificates that we receive from other nodes are
+          # authentic.
+          echo "Configuring IPsec keys"
+
+          # If the certificate does not exist or it will expire in the next 6 months
+          # (15770000 seconds), we will generate a new one.
+          if [ ! -e /etc/openvswitch/keys/ipsec-cert.pem ] || [ ! openssl x509 -noout -dates -checkend 15770000 ];
```

Comment:
shellcheck says this is invalid.  `don't wrap commands in []/[[]]`

`[ ! openssl x509 -noout -dates -checkend 15770000 ]`  doesn't seem to make sense, maybe it has never worked?


```shell
bash: [: too many arguments
```

Reply:
note that the ipsec-pre414.yaml
is exactly the script that existed there before this PR (was called ipsec.yaml)

I'll rearrange the PR so that it's a `git mv`

historically, I initially thought we wont need it anymore, but We've agreed (with @trozet) that for now we'll hold both to avoid upgrade issues

Reply:
well, seems you are right - it never worked:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-network-operator/1928/pull-ci-openshift-cluster-network-operator-master-e2e-ovn-ipsec-step-registry/1687090636676665344/artifacts/e2e-ovn-ipsec-step-registry/gather-extra/artifacts/pods/openshift-ovn-kubernetes_ovn-ipsec-9kpnd_ovn-keys.log